test/controllers/update/remove.confirm.statement.controller.spec.ts

```diff
@@ -0,0 +1,63 @@
+jest.mock("../../../src/utils/logger");
+jest.mock('../../../src/middleware/authentication.middleware');
+jest.mock('../../../src/middleware/service.availability.middleware');
+jest.mock('../../../src/utils/application.data');
+
+import { NextFunction, Request, Response } from "express";
+import request from "supertest";
+import * as config from "../../../src/config";
+import app from "../../../src/app";
+import { logger } from "../../../src/utils/logger";
+import { getApplicationData, getRemove } from "../../../src/utils/application.data";
+import {
+  ANY_MESSAGE_ERROR,
+  PAGE_TITLE_ERROR,
+  REMOVE_CONFIRMATION_STATEMENT_TITLE,
+  SERVICE_UNAVAILABLE
+} from "../../__mocks__/text.mock";
+import { serviceAvailabilityMiddleware } from "../../../src/middleware/service.availability.middleware";
+import { authentication } from "../../../src/middleware/authentication.middleware";
+import { ErrorMessages } from "../../../src/validation/error.messages";
+
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
+mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockLoggerDebugRequest = logger.debugRequest as jest.Mock;
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockGetRemove = getRemove as jest.Mock;
+
+describe("Remove confirmation statement controller", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockGetRemove.mockReturnValue({});
+  });
+
+  describe ("GET tests", () => {
+    test(`renders the ${config.REMOVE_CONFIRM_STATEMENT_PAGE}`, async () => {
```

Comment:
the appData is passed into the template and the entity name and number are rendered on the screen (taken from appData). Could you put dummy values into the mock appData and check those dummy values are in the resp.text

Reply:
Created dummy values and checks fors them.